src/components/SnoozeItem.js

```diff
@@ -74,18 +76,30 @@ function SnoozeItem({ user, snooze, onDelete }) {
           >
             {url}
           </Link>
-          {status === SNOOZE_STATUS_DONE ? (
-            <IconButton color="secondary" onClick={handleDelete}>
-              <CancelIcon />
-            </IconButton>
-          ) : (
-            <DialogButton
-              icon={<CancelIcon />}
-              title="Attention"
-              description="Do you want to delete this snooze?"
-              onConfirm={handleDelete}
-            />
-          )}
+          <Box sx={{ marginLeft: 1, display: 'flex' }}>
+            {status === SNOOZE_STATUS_DONE &&
+              <DialogFormButton
+                title="When do you want to be notified?"
+                description={url}
+                placeholder="Select or type your own value"
+                onConfirm={(notifyDate) => onUpdateSnooze({ notifyDate, snooze })}
+                size="small"
+                iconButtonComponent={<EditIcon />}
+              />
+            }
+            {status === SNOOZE_STATUS_DONE ? (
```

Comment:
what if you make a dedicated component for this block?

Reply:
On it